### PR TITLE
docs: Document access token creation and usage

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

# Changes

This PR documents how to create a token suitable for automation - i.e. an "access token"

### CODE REVIEW

File: docs/access-token.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Access Token
+
+If you want to automate tasks with the Argo Server API or CLI, you will need an access token. 
+
+Firstly, create a role with minimal permissions. This example role for jenkins only permission to update and list workflows:
+
+```yaml
```
Comment (whynowy) [inline docs/access-token.md:7]: People want to do less things, so I think giving an easy option (or both) of `kubectl create role jenkins-role --verb=list,update --resource=orkflows.argoproj.io` might be better - don't need to copy, paste and save the yaml file.

File: docs/access-token.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  - apiGroups:
+      - ""
+    resources:
+      - workflows
+    verbs:
+      - list
+      - update
+```
+
+Create a service account for your service:
```
Comment (whynowy) [inline docs/access-token.md:22]: Same here.

File: docs/access-token.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: jenkins
+```
+
+Bind the service account to the role:
+
+```yaml
```
Comment (whynowy) [inline docs/access-token.md:33]: Same here, a CLI approach.

File: docs/access-token.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: jenkins
+subjects:
+  - kind: ServiceAccount
+    name: jenkins
+```
+
+Create a secret:
```
Comment (whynowy) [inline docs/access-token.md:47]: Not sure if this is intentional to create an extra secret for the SA. 

Maybe giving a way of using default SA secret, and also mentioning that extra secrets for same SA can be created (for different clients).

Comment (alexec) [reply docs/access-token.md:47]: I wanted to proved this as as simple way to change tokens. Do you have suggestion?

File: docs/access-token.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+Firstly, create a role with minimal permissions. This example role for jenkins only permission to update and list workflows:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: jenkins-role
+rules:
+  - apiGroups:
+      - ""
```
Comment (whynowy) [inline docs/access-token.md:14]: Incorrect `apiGroups`

File: docs/access-token.md (hunk 4 above)
Comment (whynowy) [reply docs/access-token.md:47]: Rotate the default token is even easier - just delete the secret. 

Rotating extra token requires `delete` and `create`.

File: docs/access-token.md (hunk 1 above)
Comment (alexec) [reply docs/access-token.md:7]: I'm so used to GitOps :(

File: docs/access-token.md (hunk 4 above)
Comment (whynowy) [reply docs/access-token.md:23]: BTW - I don't have strong opinion on this, you decide it.

Comment (alexec) [reply docs/access-token.md]: I think you're idea is better. I've added it.

File: docs/access-token.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+Bind the service account to the role (in this case in the `argo` namespace):
+
+```shell script
+kubectl create rolebinding jenkins --role=jenkins --serviceaccount=argo:jenkins
+```
+
+You now need to get a token:
+
+```shell script
+SECRET=$(kubectl get secret  -o name | grep '^secret/jenkins-token-' | sed 's/secret\///')
```
Comment (whynowy) [inline docs/access-token.md:26]: Need to decode the token.

A workable script:

```
ARGO_TOKEN=$(kubectl -n argo get secret `kubectl -n argo get sa jenkins -o=jsonpath='{.secrets[0].name}'` -o=jsonpath='{.data.token}' | base64 --decode)
```

Comment (alexec) [reply docs/access-token.md]: Nice - really good. Thank you.

**Pull request merged**